### PR TITLE
Port pdb debugger to this repository

### PR DESCRIPTION
**Pull request by rchiodo**
Moved the code I had in https://github.com/microsoft/vscode-wasm/pull/27 to this repository instead.

@dbaeumer I didn't wire up the web version of the debugger, and the desktop version is not using a WASI, but figured that would happen later once you get the new file descriptor setup.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -95,6 +96,58 @@
...
 			}
-		}
+		},
+		"debuggers": [
+			{
+			  "label": "Python PDB",
+			  "languages": [
+				"python"
+			  ],
+			  "type": "python-pdb-node",
```
Comment (rchiodo) [inline package.json:106]: Added a debug config so I could just test with a launch.json instead of a command.

File: src/common/debugAdapter.ts [hunk 2]
```diff
@@ -6,147 +6,856 @@
...
-		this.context = context;
-		this.sequenceNumber = 1;
-		this._sendMessage = new vscode.EventEmitter();
-		this.onDidSendMessage = this._sendMessage.event;
+	constructor(
+		readonly session: vscode.DebugSession,
+		readonly context: vscode.ExtensionContext,
+		private readonly _ral: RAL
+	) {
+		this._spawner = _ral.spawner.create();
```
Comment (rchiodo) [inline src/common/debugAdapter.ts:50]: Debugger is using a 'Spawner' and 'Spawnee' abstraction. It needs only a few things from child_process.

Comment (rchiodo) [reply src/common/debugAdapter.ts:50]: Once the WASI version is working, I hope this should be easy to implement there.

File: src/common/spawner.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+
+export interface Spawner {
+	/**
+	 * Spawn a python `process`
+	 *
+	 * @param args: arguments to pass to the python
+	 * @returns an object that can be listened to for stdout/stderr/stdin
+	 */
+	spawn(args: string[], cwd: string | undefined): Promise<Spawnee>;
```
Comment (rchiodo) [inline src/common/spawner.ts:20]: I think? you can pass args to a worker? If not a message could be created to pass the args to pdb inside the worker.

**Comment by dbaeumer** [comment]
> I think? you can pass args to a worker? If not a message could be created to pass the args to pdb inside the worker.

 Yes, in node but not in the browser. In the browser you need to decode them as a param string. See https://github.com/microsoft/vscode-wasm/blob/74b49f8c2df9ecb807a934ba0de86f6fb1db5b92/sync-api-common/src/browser/ril.ts#L74 and https://github.com/microsoft/vscode-wasm/blob/74b49f8c2df9ecb807a934ba0de86f6fb1db5b92/sync-api-common/src/node/ril.ts#L74

**Comment by rchiodo** [comment]
> The code looks fine to me. I would only like to ask for one thing: if methods are private we should prefix them with the private keyword in TS and not only rely on the underscore. We don't have a clear rule in VS Code but most of the devs (including me) don't prefix private members with `_` but only rely on the visiblity modifiers provided by TS.
> 
> I propose that I add you as a committer to the repository so that you can merge the PR when approved.

Sure sounds good.

**Comment by dbaeumer** [comment]
Code got merge from another PR. Closing

**Pull request closed without merging**